src/Akka.Persistence.Sql/Query/Dao/BaseByteReadArrayJournalDao.cs

```diff
@@ -239,35 +233,59 @@ private static int MaxTake(long max)
             var maxTake = MaxTake(max);
 
             return AsyncSource<JournalRow>.FromEnumerable(
-                new { _connectionFactory = ConnectionFactory, maxTake, maxOffset, offset },
-                async input =>
+                new {_dbStateHolder , args=new QueryArgs(offset,maxOffset,maxTake) },
+                static async input =>
                 {
-                    return await input._connectionFactory.ExecuteWithTransactionAsync(
-                        ReadIsolationLevel,
-                        ShutdownToken,
-                        async (connection, token) =>
-                        {
-                            var query = connection
-                                .GetTable<JournalRow>()
-                                .Where(
-                                    r =>
-                                        r.Ordering > input.offset &&
-                                        r.Ordering <= input.maxOffset &&
-                                        r.Deleted == false)
-                                .OrderBy(r => r.Ordering)
-                                .Take(input.maxTake);
-
-                            return await AddTagDataIfNeededAsync(query, connection, token);
-                        });
+                    return await ExecuteEventQuery(input._dbStateHolder, input._dbStateHolder.Mode, input.args);
                 }
             ).Via(_deserializeFlow);
         }
+        
+        
+        internal static async Task<List<JournalRow>> ExecuteEventQuery(DbStateHolder stateHolder, TagMode tagMode, QueryArgs queryArgs)
+        {
+            return tagMode != TagMode.TagTable
```

Comment:
This is ugly and IDK how I feel but it's the easiest way to get more captures out of the way.

Reply:
> This is ugly 

Use an `if` statement instead of a ternary operator and it'll look better, I swear :p